iroh/tests/provide.rs

```diff
@@ -877,7 +877,7 @@ async fn test_token_passthrough() -> Result<()> {
     let rt = test_runtime();
     let expected = b"hello".to_vec();
     let (db, hash) = create_test_db([("test", expected.clone())]);
-    let addr = "0.0.0.0:0".parse().unwrap();
+    let addr = SocketAddr::from((Ipv4Addr::LOCALHOST, 0));
```

Comment:
this is just drive by boyscouting, right?

Reply:
not sure what you mean, but this is not crucial to the fix as the description says.  it is an orthogonal fix because I think no test should ever open public ports

Reply:
But this is no change, no? It's just a nicer way to write the same thing without parse and unwrap.